source.lua

```diff
@@ -54,31 +77,54 @@ function AddLiquidSource(pos, liquid, amount)
             end
         end
     end)
+    persist_state(g_sources_list)
 end
 
-function DeleteLiquidSource(pos)
-    for k, v in pairs(liquidSources) do
-        if same_xyz(pos, v.pos) then liquidSources[k] = nil end
-        return
+local function delete_source_at(key)
+    local v = g_sources_list[key]
+
+    if v then
+        local block = dfhack.maps.getTileBlock(v.pos)
+        if block and is_flow_passable(v.pos) then
+            local flags = dfhack.maps.getTileFlags(v.pos)
+            flags.flow_size = 0
+            dfhack.maps.enableBlockUpdates(block, true)
+        end
+        g_sources_list[key] = nil
     end
+    persist_state(g_sources_list)
+    load_liquid_source()
```

Comment:
neither of these lines should be necessary. you'll be persisting state later, and you don't need to register the repeat when deleting a source from the list